synapse/handlers/message.py

```diff
@@ -790,7 +794,10 @@ async def create_new_client_event(
             Tuple of created event, context
         """
 
+        overriding_prev_events = False
         if prev_event_ids is not None:
+            overriding_prev_events = True
```

Comment:
TODO: Is the assumption of anytime we pass in `prev_event_ids`, we use same depth good enough? What corner cases are there for places where we pass in `prev_event_ids` but it's not an query parameter override. Might need to have this where to override logic where we define that query param 🤔 

I see that we also pass in `prev_event_ids` this in [`synapse/handlers/room_member.py#L208)`](https://github.com/matrix-org/synapse/blob/8f75bf1df7f2bcb3ffe0bb89f8fe3351a48769c0/synapse/handlers/room_member.py#L208) so need to make sure that still work as expected.

Reply:
Added `inherit_depth` parameter that we pass down the line if the `prev_event` query param is used to send a message.

This was we don't interfere with all of the other scenarios that use `prev_event_ids`